app/Foliage/CmdBuild.hs

```diff
@@ -113,9 +113,22 @@ buildAction
             cf <- prepareIndexPkgCabal pkgId cabalFileTimestamp originalCabalFilePath
 
             -- all revised cabal files, with their timestamp
-            revcf <- for cabalFileRevisions $ uncurry (prepareIndexPkgCabal pkgId)
-
-            return $ cf : revcf
+            revcf <-
+              foldM
```

Comment:
Are we sure that `foldM` goes in the direction we want? I don't actually know if it's a left or right fold off the top of my head, might be better to use a more specific function.

Reply:
This would be much easier to trust if we had tests :sweat_smile: